samples/Common.c

```diff
@@ -766,6 +771,89 @@ STATUS logSignalingClientStats(PSignalingClientMetrics pSignalingClientMetrics)
+                        (DOUBLE) currentMeasureDuration;
+                    DLOGD("Packet receive rate: %lf pkts/sec", averageNumberOfPacketsReceivedPerSecond);
+
+                    outgoingBitrate = (DOUBLE)((rtcMetrics.rtcStatsObject.iceCandidatePairStats.bytesSent -
+                                                pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevNumberOfBytesSent) *
+                                               8.0) /
+                        currentMeasureDuration;
+                    DLOGD("Outgoing bit rate: %lf bps", outgoingBitrate);
+
+                    incomingBitrate = (DOUBLE)((rtcMetrics.rtcStatsObject.iceCandidatePairStats.bytesReceived -
+                                                pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevNumberOfBytesReceived) *
+                                               8.0) /
+                        currentMeasureDuration;
+                    DLOGD("Incoming bit rate: %lf bps", incomingBitrate);
+
+                    averagePacketsDiscardedOnSend =
+                        (DOUBLE)(rtcMetrics.rtcStatsObject.iceCandidatePairStats.packetsDiscardedOnSend -
+                                 pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevPacketsDiscardedOnSend) /
+                        (DOUBLE) currentMeasureDuration;
+                    DLOGD("Packet discard rate: %lf pkts/sec", averagePacketsDiscardedOnSend);
+
+                    DLOGD("Current STUN request round trip time: %lf sec", rtcMetrics.rtcStatsObject.iceCandidatePairStats.currentRoundTripTime);
+                    DLOGD("Number of STUN responses received: %llu", rtcMetrics.rtcStatsObject.iceCandidatePairStats.responsesReceived);
+
+                    prevTs = rtcMetrics.timestamp;
+                    pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevNumberOfPacketsSent =
+                        rtcMetrics.rtcStatsObject.iceCandidatePairStats.packetsSent;
+                    pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevNumberOfPacketsReceived =
+                        rtcMetrics.rtcStatsObject.iceCandidatePairStats.packetsReceived;
+                    pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevNumberOfBytesSent =
+                        rtcMetrics.rtcStatsObject.iceCandidatePairStats.bytesSent;
+                    pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevNumberOfBytesReceived =
+                        rtcMetrics.rtcStatsObject.iceCandidatePairStats.bytesReceived;
+                    pSampleConfiguration->sampleStreamingSessionList[i]->rtcMetricsHistory.prevPacketsDiscardedOnSend =
+                        rtcMetrics.rtcStatsObject.iceCandidatePairStats.packetsDiscardedOnSend;
+                }
+            }
+        }
+        THREAD_SLEEP(SAMPLE_STATS_DURATION);
```

Comment:
This will mean that on termination we have to block for up-to 60 seconds in order to clean it properly.

Instead, I recommend using TimerQueue construct and schedule periodic tasks. Once we have the timer queue, we can use the timer queue to schedule the audio and video and avoid having separate threads.

Reply:
Great suggestion! I have modified the code to use Timer queue.

Reply:
Great suggestion! I have modified the code to use timer queues. It is useful because I can use it to expand on other metrics as well in the future.